### PR TITLE
Footguns

### PR DESCRIPTION
**Pull request by iaindillingham**
Document some footguns of _not_ using cohort-joiner. These are also in the documentation, but in a section that relates specifically to ethnicity (opensafely/documentation#660).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -76,9 +86,14 @@ generate_measures:
...
 ```
 
-[1]: https://github.com/opensafely-actions/cohort-joiner/tags
-[cohort-extractor]: https://docs.opensafely.org/actions-cohortextractor/
-
 ## Notes for developers
 
 Please see [_DEVELOPERS.md_](DEVELOPERS.md).
+
+[1]: https://github.com/opensafely-actions/cohort-joiner/tags
```
Comment (milanwiedemann) [inline README.md:96]: ohh this way of creating links is soo nice :) I didnt know this works, will use this too in future

File: README.md [hunk 2]
```diff
@@ -5,7 +5,17 @@ Typically, the left-hand cohorts are several weekly or monthly extracts of pseud
 The right-hand cohort is a single extract of these data for variables that are not expected to change.
 Extracting and joining the cohorts in this way is more efficient than several weekly or monthly extracts of these data for all variables.
 
-[left join]: https://en.wikipedia.org/wiki/Join_(SQL)#Left_outer_join
+Using cohort-joiner has several advantages over writing a [scripted action][]:
+
+* cohort-joiner uses the same logic as cohort-extractor to save files,
```
Comment (milanwiedemann) [inline README.md]: nice! all advantages are clear to me. we could highlight the main point of each advantage by adding a "heading before describing it e.g.,

- Compatibility: ...
- Efficiency: ...
- Transparency: ...

File: README.md [hunk 3]
```diff
@@ -46,7 +56,7 @@ generate_ethnicity_cohort:
       cohort: output/input_ethnicity.csv
 ```
 
-The following cohort-joiner reusable action joins the cohorts, and then saves them to an output directory.
+The following cohort-joiner reusable action joins the cohorts, and then saves them to a new output directory.
```
Comment (milanwiedemann) [inline README.md:62]: what do you think about adding "as a single file"? ... and saves them (as a single file) to a new output directory ...

**Review by milanwiedemann — APPROVED** [review]
all very clear to me, added some suggestions but feel free to ignore

File: README.md (hunk 1 above)
Comment (iaindillingham) [reply README.md:96]: It's tidier than inline links, when they are long. More at https://commonmark.org/help/.

File: README.md (hunk 2 above)
Comment (iaindillingham) [reply README.md]: Don't forget the [`suggestion` block][1]! I've summarized each advantage, as you suggest, in 4d8d87e.

[1]: https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/commenting-on-a-pull-request#adding-line-comments-to-a-pull-request

File: README.md (hunk 3 above)
Comment (iaindillingham) [reply README.md:62]: I think that could be confusing. cohort-joiner saves each left-hand/right-hand pair as a single file, but there will be multiple single files when there are multiple left-hand files.

**Pull request merged**